### DISCUSSION TITLE
为什么要把`*.cls`上传到仓库中呢？

### DISCUSSION THREAD

**Discussion by YDX-2147483647** [BITNP/BIThesis#380] [General]
这些文件由`*.dtx`生成，并且有很强重复性，干扰`git diff`。合并分支时，即使能自动合并，也很可能要重新`make && make copy`更新`*.cls`。如果只是发布时带上，就没有这些问题了。

另外，现在除了 graduate-thesis，其它`*.cls`都被忽略了，只是仓库里没删。

```shell
$ rg cls --glob **/.gitignore
templates\graduate-thesis\.gitignore
316:!*.cls

.gitignore
14:*.cls
```

> 拆分自 #359。

---

更新：#381 正着手删除。

**Comment by fky2015** [comment]
算是 https://github.com/BITNP/BIThesis/discussions/100 的遗留问题。

只有先改 CI 才能将他们删除。

**Reply by YDX-2147483647** [reply]
原来如此。

**Reply by fky2015** [reply]
之所以 CI 这边不好移除的原因是：
我们在 Github Actions 里面用了 matrix，因此没有办法判断哪个 template 应该添加哪个 `*.cls`。
